src/software/proto/message_translation/ssl_message_translator.h

```diff
@@ -0,0 +1,180 @@
+    {SSLFieldLines::RIGHT_GOAL_TOP_LINE, "RightGoalTopLine"},
+    {SSLFieldLines::RIGHT_GOAL_BOTTOM_LINE, "RightGoalBottomLine"},
+    {SSLFieldLines::RIGHT_GOAL_DEPTH_LINE, "RightGoalDepthLine"},
+    {SSLFieldLines::LEFT_GOAL_TOP_LINE, "LeftGoalTopLine"},
+    {SSLFieldLines::LEFT_GOAL_BOTTOM_LINE, "LeftGoalBottomLine"},
+    {SSLFieldLines::LEFT_GOAL_DEPTH_LINE, "LeftGoalDepthLine"},
+    {SSLFieldLines::LEFT_FIELD_LEFT_PENALTY_STRETCH, "LeftFieldLeftPenaltyStretch"},
+    {SSLFieldLines::LEFT_FIELD_RIGHT_PENALTY_STRETCH, "LeftFieldRightPenaltyStretch"},
+    {SSLFieldLines::RIGHT_FIELD_LEFT_PENALTY_STRETCH, "RightFieldLeftPenaltyStretch"},
+    {SSLFieldLines::RIGHT_FIELD_RIGHT_PENALTY_STRETCH, "RightFieldRightPenaltyStretch"},
+};
+
+MAKE_ENUM(SSLCircularArcs, CENTER_CIRCLE)
+
+static const std::map<SSLCircularArcs, const std::string> ssl_circular_arc_names = {
+    {SSLCircularArcs::CENTER_CIRCLE, "CenterCircle"},
+};
+
+/**
+ * Find an SSL FieldLineSegment by name. Returns the first instance of a line segment
+ * with the given name if it exists, otherwise returns std::nullopt
+ *
+ * @param line_segments The line segments to search
+ * @param name The name of the segment to find
+ *
+ * @return The first line segment with the given name if it exists, otherwise
+ * returns std::nullopt
+ */
+std::optional<SSL_FieldLineSegment> findLineSegment(
+    const google::protobuf::RepeatedPtrField<SSL_FieldLineSegment>& line_segments,
+    const std::string& name);
+
+/**
+ * Find an SSL FieldCircularArc by name. Returns the first instance of a field arc
+ * with the given name if it exists, otherwise returns std::nullopt
+ *
+ * @param circular_arcs The arcs to search
+ * @param name The name of the arc to find
+ *
+ * @return The first arc with the given name if it exists, otherwise
+ * returns std::nullopt
+ */
+std::optional<SSL_FieldCircularArc> findCircularArc(
+    const google::protobuf::RepeatedPtrField<SSL_FieldCircularArc>& circular_arcs,
+    const std::string& name);
+
+/**
+ * Creates a Vector2f proto message from the given point
+ *
+ * @param point The point to convert to a Vector2f
+ *
+ * @return A Vector2f message with the same coordinates as the given point
+ */
+std::unique_ptr<Vector2f> createVector2f(const Point& point);
+
+/**
+ * Creates an SSL_FieldLineSegment proto message
+ *
+ * @pre thickness must be >= 0
+ *
+ * @throws std::invalid_argument if thickness < 0
+ *
+ * @param segment The segment geometry the message with contain
+ * @param thickness The thickness of the line in metres
+ * @param name The name of the line segment
+ * @param shape_type The type of field shape this segment represents
+ *
+ * @return an SSL_FieldLineSegment representing the given segment with the
+ * given thickness
+ */
+std::unique_ptr<SSL_FieldLineSegment> createFieldLineSegment(
+    const Segment& segment, float thickness, const std::string& name,
+    const SSL_FieldShapeType& shape_type);
```

Comment:
Similar comment here about using raw strings instead of an enum

Reply:
Similar comment applies for function(s) below.

Reply:
done for all functions that previously took a string